packages/common/src/utils/filters.ts

```diff
@@ -421,6 +422,133 @@ export const getFilterRulesByFieldType = (
+                        ...accumulator.dimensions,
+                        [getFilterGroupItemsPropertyName(flatFilterGroup)]: [
+                            ...getItemsFromFilterGroup(accumulator.dimensions),
+                            item,
+                        ],
+                    } as FilterGroup;
+                } else if (isTableCalculationField(fieldInRule)) {
+                    accumulator.tableCalculations = {
+                        id: uuidv4(),
+                        ...accumulator.tableCalculations,
+                        [getFilterGroupItemsPropertyName(flatFilterGroup)]: [
+                            ...getItemsFromFilterGroup(
+                                accumulator.tableCalculations,
+                            ),
+                            item,
+                        ],
+                    } as FilterGroup;
+                } else {
+                    accumulator.metrics = {
+                        id: uuidv4(),
+                        ...accumulator.metrics,
+                        [getFilterGroupItemsPropertyName(flatFilterGroup)]: [
+                            ...getItemsFromFilterGroup(accumulator.metrics),
+                            item,
+                        ],
+                    } as FilterGroup;
+                }
+            }
+        }
+
+        if (isFilterGroup(item)) {
+            const filters = getFiltersFromGroup(item, fields);
+
+            if (filters.dimensions) {
+                accumulator.dimensions = {
+                    id: uuidv4(),
+                    ...accumulator.dimensions,
+                    [getFilterGroupItemsPropertyName(accumulator.dimensions)]: [
+                        ...getItemsFromFilterGroup(accumulator.dimensions),
+                        filters.dimensions,
+                    ],
+                } as FilterGroup;
+            }
+
+            if (filters.metrics) {
+                accumulator.metrics = {
+                    id: uuidv4(),
+                    ...accumulator.metrics,
+                    [getFilterGroupItemsPropertyName(accumulator.metrics)]: [
+                        ...getItemsFromFilterGroup(accumulator.metrics),
+                        filters.metrics,
+                    ],
+                } as FilterGroup;
+            }
+
+            if (filters.tableCalculations) {
+                accumulator.tableCalculations = {
+                    id: uuidv4(),
+                    ...accumulator.tableCalculations,
+                    [getFilterGroupItemsPropertyName(
+                        accumulator.tableCalculations,
+                    )]: [
+                        ...getItemsFromFilterGroup(
+                            accumulator.tableCalculations,
+                        ),
+                        filters.tableCalculations,
+                    ],
+                } as FilterGroup;
+            }
+        }
+
+        return accumulator;
+    }, {} as Filters);
+};
+
```

Comment:
Will need to make this function a bit more readable as here is where most of the logic is, basically it converts the filter groups into filters of type `Filters` with the types associated with them so they can be saved in the database

Reply:
nvm, spoke with @owlas we can keep this as is since and refactor in case there's any performance hits because of it